### PR TITLE
better recosting

### PR DESCRIPTION
**Pull request by CuriOusQuOkka**
# Issue

What issue is this PR targeting? If there is no issue that addresses the problem, please open a corresponding issue and link it here.

## Tasklist

 - [ ] Add tests
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Update the docs with any new request parameters or changes to behavior described
 - [ ] Update the [changelog](CHANGELOG.md)
 - [ ] If you made changes to the lua files, update the [taginfo](taginfo.json) too.

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: scripts/valhalla_build_config [hunk 1]
```diff
@@ -98,7 +98,8 @@ config = {
     },
     'max_reserved_labels_count': 1000000,
     'clear_reserved_memory': False,
-    'extended_search': False
+    'extended_search': False,
+    'invariant_recosting': True
```
Comment (kevinkreiser) [inline scripts/valhalla_build_config:102]: i think this should be decided at runtime, not via config. simply add a new field to the request structure that tells valhalla to use bidirectional a* even if the date_time type isnt invariant and  you're basically done (other than the actual work of doing the linear combination in GetSpeed and passing the seconds_from_now in the costing function)

File: src/sif/recost.cc [hunk 2]
```diff
@@ -122,7 +122,8 @@ void recost_forward(baldr::GraphReader& reader,
     // update the cost to the end of this edge
     uint8_t flow_sources;
     cost += transition_cost +
-            costing.EdgeCost(edge, tile, offset_time.second_of_week, flow_sources) * edge_pct;
+            costing.EdgeCost(edge, tile, offset_time.second_of_week, flow_sources, seconds_offset) *
```
Comment (kevinkreiser) [inline src/sif/recost.cc:125]:
```suggestion
            costing.EdgeCost(edge, tile, offset_time.second_of_week, flow_sources, offset_time.seconds_from_now) *
```

File: scripts/valhalla_build_config (hunk 1 above)
Comment (kevinkreiser) [reply scripts/valhalla_build_config:102]: sorry maybe a little more info, you'd need to account for this in thor/route_action where we select the algorithm to use.

**Comment by CuriOusQuOkka** [comment]
it was a draft branch, see original one here https://github.com/valhalla/valhalla/pull/3398

**Pull request closed without merging**